registry/storage/driver/s3/s3.go

```diff
@@ -587,6 +587,11 @@ func (d *driver) List(path string) ([]string, error) {
 	if path != "/" && path[len(path)-1] != '/' {
 		path = path + "/"
 	}
+	prefix := ""
+	if d.s3Path("") == "" {
```

Comment:
Because everywhere else it is used, it shouldn't. In fact it's used here only as a proxy for the root directory. I'll clean it up in a bit.